src/htcondor_es/convert_to_json.py

```diff
@@ -1146,18 +1146,38 @@ def commonExitCode(ad):
     """
     Consolidate the exit code values of JobExitCode,
     the  chirped CRAB and WMCore values, and
-    the original condor exit code.
-    JobExitCode and Chirp_CRAB3_Job_ExitCode
-    exists only on analysis jobs.
+    the original condor exit code, according to 
+    the workflow type: production or analysis.
+
+    Input parameter: ad #this is the raw ad from condor
     """
-    return ad.get(
-        "JobExitCode",
-        ad.get(
-            "Chirp_CRAB3_Job_ExitCode",
-            ad.get("Chirp_WMCore_cmsRun_ExitCode", ad.get("ExitCode", 50666)),
-        ),
-    )
+    # If the raw ExitCode in the ad is not present,
+    # condor did not finish, hence we return 50666
```

Comment:
I feel like this "condor did not finish" sentence could be misleading. If my memory does not fail me, the lack of `ExitCode` in the job classad means that the job was actually removed from condor (without gracefully finishing). If that is correct, maybe we should rephrase it around that.

Reply:
Fair enough, I have changed the commend to be more clear.